examples/nextjs-spa/pages/login.tsx

```diff
@@ -0,0 +1,136 @@
+// React
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+// Ory SDK
+import { ory } from "../components/sdk"
+import { LoginFlow, UpdateLoginFlowBody } from "@ory/client"
+
+import { AxiosError } from "axios"
+import type { NextPage } from "next"
+
+import { UserAuthCard } from "@ory/elements"
+
+import React from "react"
+import Link from "next/link"
+
+const Login: NextPage = () => {
+  const [flow, setFlow] = useState<LoginFlow | null>(null)
+
+  // Get ?flow=... from the URL
+  const router = useRouter()
+  const {
+    return_to: returnTo,
+    flow: flowId,
+    // Refresh means we want to refresh the session. This is needed, for example, when we want to update the password
+    // of a user.
+    refresh,
+    // AAL = Authorization Assurance Level. This implies that we want to upgrade the AAL, meaning that we want
+    // to perform two-factor authentication/verification.
+    aal,
+  } = router.query
+
+  // This might be confusing, but we want to show the user an option
+  // to sign out if they are performing two-factor authentication!
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return
+    }
+
+    // If ?flow=.. was in the URL, we fetch it
+    if (flowId) {
+      ory
+        .getLoginFlow({ id: String(flowId) })
+        .then(({ data }) => {
+          setFlow(data)
+        })
+        .catch(
+          (error: AxiosError) =>
+            // If the flow was not found, we redirect to the login page
+            error.response?.status === 404 && router.push("/login"),
+        )
+      return
+    }
+
+    // Otherwise we initialize it
+    ory
+      .createBrowserLoginFlow({
+        refresh: Boolean(refresh),
+        aal: aal ? String(aal) : undefined,
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        setFlow(data)
+      })
+      .catch(
+        (error: AxiosError) =>
+          // If the flow was not found, we redirect to the login page
+          error.response?.status === 404 && router.push("/login"),
```

Comment:
why 404 redirect to login again?

404 probably means the sdk url is not set correctly. Could we not ask the user to check this on an error page?